### PR TITLE
feat(content-sidebar): call handlers with initial panel and open state

### PR DESCRIPTION
**Pull request by kkuliczkowski-box**
### Description
Existing `onPanelChange` and `onOpenChange` prop handlers were extended to accept a second parameter `isInitialState`, along with logic changes, in order to notify parent about the initial open state and initially rendered panel.

### CODE REVIEW

**Comment by karolinaru** [comment]
I don't have any comments on the PR, but just one general question – I noticed you're using `noop` in a few places. I'm curious, what's the reason behind that? Is it considered good practice to include it?

**Comment by kkuliczkowski-box** [comment]
> I don't have any comments on the PR, but just one general question – I noticed you're using `noop` in a few places. I'm curious, what's the reason behind that? Is it considered good practice to include it?

I think that's one way of approaching optional handler props, so there's no need to include a condition check before calling them.

File: src/elements/content-sidebar/Sidebar.js [hunk 1]
```diff
@@ -107,11 +107,13 @@ class Sidebar extends React.Component<Props, State> {
...
 
     componentDidMount() {
-        const { file, api, metadataSidebarProps, docGenSidebarProps }: Props = this.props;
+        const { file, api, metadataSidebarProps, docGenSidebarProps, onOpenChange = noop }: Props = this.props;
         // if docgen feature is enabled, load metadata to check whether file is a docgen template
         if (docGenSidebarProps.enabled) {
             docGenSidebarProps.checkDocGenTemplate(api, file, metadataSidebarProps.isFeatureEnabled);
         }
+        const isOpen = this.isOpen();
+        onOpenChange(isOpen, true);
```
Comment (marcoartaviaq) [inline src/elements/content-sidebar/Sidebar.js:116]: nit: can we pass `this.isOpen` directly to the function without creating a new const?

File: src/elements/content-sidebar/SidebarNav.js [hunk 2]
```diff
@@ -77,7 +82,7 @@ const SidebarNav = ({
                             data-resin-target={SIDEBAR_NAV_TARGETS.BOXAI}
                             data-target-id="SidebarNavButton-boxAI"
                             data-testid="sidebarboxai"
-                            onClick={onPanelChange}
```
Comment (marcoartaviaq) [inline src/elements/content-sidebar/SidebarNav.js:80]: Why this change was necessary?

File: src/elements/content-sidebar/SidebarPanels.js [hunk 3]
```diff
@@ -236,6 +249,7 @@ class SidebarPanels extends React.Component<Props, State> {
                         exact
                         path={`/${SIDEBAR_VIEW_BOXAI}`}
                         render={() => {
+                            this.handlePanelRender(SIDEBAR_VIEW_BOXAI);
```
Comment (bfoxx1906) [inline src/elements/content-sidebar/SidebarPanels.js:252]: What happens if the LoadableBoxAISidebar does not load for some reason? Do we still want to call the call back before the panel loads like this?

File: src/elements/content-sidebar/Sidebar.js [hunk 4]
```diff
@@ -61,8 +61,8 @@ type Props = {
     metadataEditors?: Array<MetadataEditor>,
     metadataSidebarProps: MetadataSidebarProps,
     onAnnotationSelect?: Function,
-    onOpenChange?: (isOpen: boolean) => void,
-    onPanelChange?: (name: string) => void,
+    onOpenChange?: (isOpen: boolean, isInitialState?: boolean) => void,
```
Comment (bfoxx1906) [inline src/elements/content-sidebar/Sidebar.js:64]: Would it be more intuitive to make isInitialState non optional? This seems to be an important part of this functionality and we're essentially treating not passing the optional parameter back as the false value anyway.

File: src/elements/content-sidebar/SidebarNav.js (hunk 2 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/SidebarNav.js:80]: I felt that the Button's `onClick` should not really to be aware of the handler function and because I've changed the type of `onPanelChange` I didn't want to change Button's `onClick` type.

File: src/elements/content-sidebar/SidebarPanels.js (hunk 3 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/SidebarPanels.js:252]: Actually we do as we still want to track which of the panels would be open if it loaded successfully. At this point we know which one would the active one.

File: src/elements/content-sidebar/Sidebar.js (hunk 4 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/Sidebar.js:64]: Hmm, I didn't want to change the previous implementation, hence making the second parameter optional, but I think you're right, I'll change this, so that every use is intentional passing either `false` or `true`.

Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/Sidebar.js]: Done

**Review by bfoxx1906 — APPROVED** [review]
LGTM. Thanks for the updates.

**Review by marcoartaviaq — APPROVED** [review]
